src/main/kotlin/app/revanced/patches/twitch/misc/settings/resource/patch/SettingsResourcePatch.kt

```diff
@@ -0,0 +1,133 @@
+package app.revanced.patches.twitch.misc.settings.resource.patch
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.DomFileEditor
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.ResourcePatch
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patches.shared.settings.BasePreference
+import app.revanced.patches.shared.settings.IResource
+import app.revanced.patches.shared.settings.addPreference
+import app.revanced.patches.shared.settings.addResource
+import app.revanced.patches.shared.settings.impl.ArrayResource
+import app.revanced.patches.shared.settings.impl.StringResource
+import app.revanced.patches.twitch.misc.settings.annotations.SettingsCompatibility
+import app.revanced.patches.youtube.misc.manifest.patch.FixLocaleConfigErrorPatch
+import app.revanced.util.resources.ResourceUtils
+import app.revanced.util.resources.ResourceUtils.copyResources
+import org.w3c.dom.Node
+
+@Name("settings-resource-patch")
+@SettingsCompatibility
+@DependsOn([FixLocaleConfigErrorPatch::class])
+@Version("0.0.1")
+class SettingsResourcePatch : ResourcePatch {
```

Comment:
This class is very similar to the `SettingsResourcePatch` class for YouTube. Probably, it is not a bad idea, to create an abstract class for this, the same way it has been done for the integrations patch for YouTube and TikTok.

Reply:
Sure, I'll take a look. `SettingsResourcePatch` has most of this logic in a companion object, so I'd probably need to create an abstract class for the companion object and one for the actual patch.

Reply:
An abstract class was just an idea, if you see a better fitting implementation to get rid of the duplicate code, then do that.